### ISSUE TITLE
CA1708 for file local types

### ISSUE THREAD

**Issue by quixoticaxis** [dotnet/roslyn-analyzers#6953]
### Analyzer

**Diagnostic ID**: [CA1708](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1708): `Identifiers should differ by more than case`

### Analyzer source

**SDK**: .Net 8 RC 1 (8.0.100-rc.1.23455.8)

**Version**: .Net 8 RC 1 (8.0.100-rc.1.23455.8)

### Describe the bug

When there are two file local classes with the same name in the same namespace, CA1708 is being triggered.

### Steps To Reproduce

1. Execute `dotnet new console --name CA1708`.
2. Add `<TreatWarningsAsErrors>true</TreatWarningsAsErrors>` and `<AnalysisMode>AllEnabledByDefault</AnalysisMode>` to .csproj.
3. Create two files: A.cs and B.cs.
```csharp
namespace CA1708;

public static class A
{
    public static void Call() => Helper.Call();
}

file static class Helper
{
    public static void Call() {}
}
```
```csharp
namespace CA1708;

public static class B
{
    public static void Call() => Helper.Call();
}

file static class Helper
{
    public static void Call() {}
}

```
5. Add the following code to Program.cs:
```csharp
using CA1708;

A.Call();
B.Call();
```
6. Execute `dotnet build`.

### Expected behavior
The project is built.
### Actual behavior
The build fails with: 

> CSC : error CA1708: Names of 'Types' and 'CA1708.Helper, CA1708.Helper' should differ by more than case (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1708)

**Comment by mpidash** [comment]
This is a duplicate of #6514.

**Comment by quixoticaxis** [comment]
> This is a duplicate of #6514.

I see, I somehow failed to find it via search. Closing.
